modular_skyrat/code/modules/mob/living/carbon/human/species_types/synthetic/ipc.dm

```diff
@@ -52,6 +62,8 @@
 		O.synthetic = TRUE
 
 /datum/species/ipc/spec_revival(mob/living/carbon/human/H)
+	if(H.stat == DEAD) //by this point we should be alive
+		return
```

Comment:
spec_revival only procs when revive() returns 1, which is when someone was successfully revived, so either an issue with revivals or people die instantly afterwards